src/commands/createApiCenterService.ts

```diff
@@ -0,0 +1,95 @@
+        if (!serverName) {
+            return;
+        }
+        const apiCenterService = new ApiCenterService(node.subscriptionContext!, serverName, serverName);
+        const serverRes = await apiCenterService.getSubServerList();
+        const locations = serverRes.resourceTypes.find((item: any) => item.resourceType === 'services')?.locations;
+        if (!locations || locations.length === 0) {
+            vscode.window.showWarningMessage(UiStrings.NoLocationAvailable);
+            return;
+        }
+        const location = await vscode.window.showQuickPick(locations, { placeHolder: UiStrings.SelectLocation });
+        if (!location) {
+            return;
+        }
+
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: UiStrings.CreateApiCenterService
+        }, async (progress, token) => {
+            progress.report({ message: UiStrings.GetResourceGroup });
+            const rgExist = await apiCenterService.isResourceGroupExist();
+            if (!rgExist) {
+                progress.report({ message: UiStrings.CreatingResourceGroup });
+                await AzureApiCenterService.confirmResourceGroupWithRetry(apiCenterService, location);
+            }
+
+            progress.report({ message: UiStrings.CreatingApiCenterService });
+            validateResponse(await apiCenterService.createOrUpdateApiCenterService(location));
+            // Retry mechanism to check API center creation status
+            await AzureApiCenterService.confirmServerStatusWithRetry(serverName, node, context);
+        });
+    };
+    export function validateResponse(response: any) {
+        if (response && (response as any).message) {
+            throw new Error((response as any).message);
+        }
+    };
+    export async function confirmResourceGroupWithRetry(apiCenterService: ApiCenterService, location: string) {
+        let retryCount = 0;
+        const maxRetries = 5;
+        const retryDelay = 10000; // 10 seconds
+        let result: ResourceGroup;
+        do {
+            validateResponse(result = await apiCenterService.createOrUpdateResourceGroup(location));
+            if (result && result.properties && result.properties.provisioningState && result.properties.provisioningState === 'Succeeded') {
+                break;
+            }
+            retryCount++;
+            await GeneralUtils.sleep(retryDelay);
+        } while (retryCount < maxRetries);
+        if (retryCount >= maxRetries) {
+            throw new Error(UiStrings.LongTimeToCreateResourceGroup);
+        }
+    };
+    export async function confirmServerStatusWithRetry(serverName: string, node: SubscriptionTreeItem, context: IActionContext) {
+        let retryCount = 0;
+        const maxRetries = 5;
+        const retryDelay = 10000; // 10 seconds
```

Comment:
Could we decrease the Delay to show creation is successful ealier?

Reply:
My suggestion is: retryDelay = 5 seconds, maxRetries = 10. (unless there is rate limit?)

For current method, the unlucky would wait for extra 32 seconds in worst case?

Reply:
update to 3s with max 10 tries.
and this is the doc share the rest call rate limit
https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling#subscription-and-tenant-limits

this policy will not reach the limit.